### ISSUE TITLE
Support for built-in extentions

### ISSUE THREAD

**Issue by hevayo** [ballerina-platform/ballerina-lang#30351]
**Description:**
Currently a compiler extension is distributed in a package and you need to import it to activate the extension. But in some cases you need to ship a compiler extension with the distribution itself and they should be engaged by default. 

As a solution we can implement a concept of "built in extensions" where the extension will be discover and leaded via a java SPI. The extensions can be added to the distribution as an when needed. 

Couple of use cases this will be useful includes
* Data validator extension used for data mapper
* CLI option parser validator 

Basically the above extensions will add additional compile time validations which cannot be included in the compiler.

**Comment by azinneera** [comment]
**Approach**
There are two types of compiler plugins: package-provided compiler plugins and built-in compiler plugins. 

package-provided plugins are already supported which can be written by extending `io.ballerina.projects.plugins.CompilerPlugin`. In-built compiler plugins are supported the same way. 
https://github.com/ballerina-platform/ballerina-lang/blob/721673cceff2ec290ac2bf4f5dc69795f6130f60/compiler/ballerina-lang/src/main/java/io/ballerina/projects/plugins/CompilerPlugin.java#L27

**Plugging a package-provided compiler plugin** 
* Created and attached to a package using the `CompilerPlugin.toml` file
* Will run for packages that imports the package containing the compiler plugin

**Plugging a built-in compiler plugin** 
* Created and packaged with the distribution in `<DIST_PATH>/bre/lib` 
* Will run for all Ballerina packages built from sources (BuildProject type projects in API terms)
